build.py

```diff
@@ -45,17 +46,49 @@ def _parse_args():
     parsed = args.parse_args()
     assert parsed.max_seq_len == -1 or parsed.max_seq_len > 0
 
-    parsed.model_path = os.path.join(parsed.artifact_path, "models", parsed.model)
     parsed.export_kwargs = {}
     parsed.lib_format = "so"
+
+    parsed = _setup_model_path(parsed)
+
     parsed.db_path = parsed.db_path or os.path.join("log_db", parsed.model)
+
     utils.parse_target(parsed)
     utils.argparse_postproc_common(parsed)
+
     parsed.artifact_path = os.path.join(
         parsed.artifact_path, f"{parsed.model}-{parsed.quantization.name}"
     )
+
     return parsed
 
+def _setup_model_path(args):
+
+    if args.model_path and args.hf_path:
+        assert (args.model_path and not args.hf_path) or (args.hf_path and not args.model_path), "You cannot specify both a model path and a HF path. Please select one to specify."
+
+    if args.model_path:
+        assert os.path.exists(os.path.join(args.model_path, "config.json")), "Model path must contain valid config file."
+        with open(os.path.join(args.model_path, "config.json")) as f:
+            config = json.load(f)
+            assert ("model_type" in config) and ("_name_or_path" in config), "Invalid config format."
+            assert config["model_type"] in utils.supported_model_types, f"Model type {config['model_type']} not supported."
+        args.model = config["_name_or_path"].split("/")[-1]
+    elif args.hf_path:
+        args.model = args.hf_path.split("/")[-1]
+        args.model_path = os.path.join(args.artifact_path, "models", args.model)
+        if os.path.exists(args.model_path):
+            print(f"Weights exist at {args.model_path}, skipping download.")
+        else:
+            os.makedirs(args.model_path, exist_ok=True)
+            os.system("git lfs install")
+            os.system(f"git clone https://huggingface.co/{args.hf_path} {args.model_path}")
+            print(f"Downloaded weights to {args.model_path}")
```

Comment:
do model type check as local model path

Reply:
Addressed in https://github.com/mlc-ai/mlc-llm/pull/140/commits/10bda9f6661d8fbf91a37493e11f9e4060dc11be.